yjit/src/codegen.rs

```diff
@@ -5265,11 +5301,117 @@ fn gen_send_general(
+                        }
+
+                        argc -= 1;
+
+                        let compile_time_name = jit_peek_at_stack(jit, ctx, argc as isize);
+
+                        mid = unsafe { rb_get_symbol_id(compile_time_name) };
+                        if mid == 0 {
+                            gen_counter_incr!(asm, send_optimized_method_send_null_mid);
+                            return CantCompile;
+                        }
+
+                        cme = unsafe { rb_callable_method_entry(comptime_recv_klass, mid) };
+                        if cme.is_null() {
+                            gen_counter_incr!(asm, send_optimized_method_send_null_cme);
+                            return CantCompile;
+                        }
+
+                        // We aren't going to handle `send(send(:foo))`. We would need to
+                        // do some stack manipulation here or keep track of how many levels
+                        // deep we need to stack manipulate
+                        // Because of how exits currently work, we can't do stack manipulation
+                        // until we will no longer side exit.
+                        let def_type = unsafe { get_cme_def_type(cme) };
+                        if let VM_METHOD_TYPE_OPTIMIZED = def_type {
+                            let opt_type = unsafe { get_cme_def_body_optimized_type(cme) };
+                            if let OPTIMIZED_METHOD_TYPE_SEND = opt_type {
+                                gen_counter_incr!(asm, send_optimized_method_send_nested);
+                                return CantCompile;
+                            }
+                        }
+
+                        let kw_splat = if flags & VM_CALL_KW_SPLAT != 0 {
+                            VM_CALL_KW_SPLAT
+                        } else {
+                            0
+                        };
+                        flags = VM_CALL_FCALL | VM_CALL_OPT_SEND | kw_splat;
+
+                        assume_method_lookup_stable(jit, ocb, comptime_recv_klass, cme);
+
+                        if !compile_time_name.string_p() && !compile_time_name.static_sym_p()  {
+                            gen_counter_incr!(asm, send_optimized_method_send_chain_not_string_or_sym);
+                            return CantCompile;
+                        }
+
+                        let name_opnd = asm.load(ctx.stack_opnd(argc));
+
+                        let (known_class, type_mismatch_exit, compile_time_symbol, symbol_opnd) = {
+                            if compile_time_name.string_p() {
+                                (
+                                    unsafe { rb_cString },
+                                    counted_exit!(ocb, side_exit, send_optimized_method_send_chain_not_string),
+                                    unsafe { rb_str_intern(compile_time_name) },
+                                    asm.ccall(rb_str_intern as *const u8, vec![name_opnd])
```

Comment:
I _think_ this can allocate if the symbol for the string we encounter at runtime doesn't already exist and because we haven't prepared for a routine call that may be unsafe? (The potential GC wouldn't know of any pushed stack values that CFP->sp hasn't been updated for)

Reply:
Good call out. It looks like it can because `USE_SYMBOL_GC` seems to be on by default. Haven't had to deal with GC yet, so don't fully understand the implications here or what I should do given this.

Reply:
@jhawthorn is there a way to request a symbol if it's interned already but otherwise return Qnil 🤔  ?

It might be good to try to create an adversarial test case that does allocate a new symbol.

Reply:
I figured out how to trigger this issue. (calling a method that does not exist will allocate). Will try out using `lookup_str_sym` instead. It returns Qfalse if the symbol doesn't exist. It does not seem to allocate. And if the symbol does not exist, that means the method does not exist, so we can just side_exit.